### PR TITLE
clippy: Address several complexity lints

### PR DESCRIPTION
**Pull request by bradjc**
### Pull Request Overview

This pull request addresses several complexity clippy lints:

```diff
diff --git a/tools/run_clippy.sh b/tools/run_clippy.sh
index 45bc69afa..6d2337366 100755
--- a/tools/run_clippy.sh
+++ b/tools/run_clippy.sh
@@ -63,20 +63,10 @@ CLIPPY_ARGS_COMPLEXITY="


 -A clippy::unnecessary_cast
--A clippy::extra_unused_lifetimes
 -A clippy::unnecessary_unwrap
 -A clippy::needless_lifetimes
--A clippy::useless_conversion
--A clippy::precedence
--A clippy::redundant_slicing
--A clippy::derivable_impls
--A clippy::char_lit_as_u8
--A clippy::needless_bool
--A clippy::useless_asref
--A clippy::clone-on-copy
 -A clippy::explicit_auto_deref
 -A clippy::explicit_counter_loop
--A clippy::manual_unwrap_or
 -A clippy::borrow_deref_ref
 -A clippy::overflow_check_conditional
 -A clippy::needless-match
```

Most of them were done automatically but removing unnecessary lifetimes was a bit more manual.

### Testing Strategy

travis

### TODO or Help Wanted

n/a

### Documentation Updated

- [x] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: capsules/extra/src/lsm303agr.rs [hunk 1]
```diff
@@ -409,11 +409,7 @@ impl<I: i2c::I2CDevice> i2c::I2CClient for Lsm303agrI2C<'_, I> {
     fn command_complete(&self, buffer: &'static mut [u8], status: Result<(), i2c::Error>) {
         match self.state.get() {
             State::IsPresent => {
-                let present = if status == Ok(()) && buffer[0] == 60 {
-                    true
-                } else {
-                    false
-                };
+                let present = status == Ok(()) && buffer[0] == 60;
```
Comment (jrvanwhy) [inline capsules/extra/src/lsm303agr.rs:412]: I personally prefer `let present = status.is_ok() && buffer[0] == 60` but this is fine too.
This also appears in `capsules/extra/src/lsm303dlhc.rs` and `capsules/extra/src/mlx90614.rs`.

File: chips/nrf52/src/ficr.rs [hunk 2]
```diff
@@ -450,25 +450,25 @@ impl Ficr {
             .read(DeviceAddress1::DEVICEADDRESS);
 
         let h: [u8; 16] = [
-            '0' as u8, '1' as u8, '2' as u8, '3' as u8, '4' as u8, '5' as u8, '6' as u8, '7' as u8,
-            '8' as u8, '9' as u8, 'a' as u8, 'b' as u8, 'c' as u8, 'd' as u8, 'e' as u8, 'f' as u8,
+            b'0', b'1', b'2', b'3', b'4', b'5', b'6', b'7', b'8', b'9', b'a', b'b', b'c', b'd',
+            b'e', b'f',
         ];
```
Comment (jrvanwhy) [inline chips/nrf52/src/ficr.rs:455]: How about:

```rust
let h: [u8; 16] = *b"0123456789abcdef";
```

**Review by jrvanwhy — APPROVED** [review]

**Review by hudson-ayers — APPROVED** [review]
Might want to force-push this to rerun the netlify build (it had a time-out)

**Comment by hudson-ayers** [comment]
Hmm, now clippy is failing

**Review by jrvanwhy — APPROVED** [review]

File: chips/imxrt10xx/src/lpi2c.rs [hunk 3]
```diff
@@ -703,9 +703,9 @@ impl<'a> Lpi2c<'a> {
 
         // setting slave address
         self.registers.mier.modify(MIER::EPIE::CLEAR);
-        self.registers
-            .mtdr
-            .write(MTDR::CMD.val(100) + MTDR::DATA.val((self.slave_address.get() << 1 + 1) as u32));
+        self.registers.mtdr.write(
+            MTDR::CMD.val(100) + MTDR::DATA.val((self.slave_address.get() << (1 + 1)) as u32),
+        );
```
Comment (brghena) [inline chips/imxrt10xx/src/lpi2c.rs:708]: This one worries me. I assume the parentheses are just demonstrating what was already the order of precedence, but it feels sketchy and I'm not sure there isn't a bug here.

File: chips/rp2040/src/gpio.rs [hunk 4]
```diff
@@ -347,13 +347,13 @@ impl<'a> RPPins<'a> {
...
-                    self.gpio_registers.intr[bank_no].set(current_val & (1 << l_low_reg_no + 2));
+                } else if (current_val & enabled_val & (1 << (l_low_reg_no + 2))) != 0 {
+                    self.gpio_registers.intr[bank_no].set(current_val & (1 << (l_low_reg_no + 2)));
                     self.pins[pin + bank_no * 8].handle_interrupt();
-                } else if (current_val & enabled_val & (1 << l_low_reg_no + 3)) != 0 {
-                    self.gpio_registers.intr[bank_no].set(current_val & (1 << l_low_reg_no + 3));
+                } else if (current_val & enabled_val & (1 << (l_low_reg_no + 3))) != 0 {
+                    self.gpio_registers.intr[bank_no].set(current_val & (1 << (l_low_reg_no + 3)));
                     self.pins[pin + bank_no * 8].handle_interrupt();
                 }
```
Comment (brghena) [inline chips/rp2040/src/gpio.rs:358]: The lack of parallelism here for the `(l_low_reg_no + 1)` case feels like a bug.

File: chips/imxrt10xx/src/lpi2c.rs (hunk 3 above)
Comment (brghena) [reply chips/imxrt10xx/src/lpi2c.rs:708]: In fact, I think there are two bugs. `100` should be `0b100` (I think), and also the last "+1" is adding to the slave address to turn it into an I2C read.

**Review by brghena — COMMENTED** [review]
Two possible bugs here, but they aren't this PR's fault. I'm opening separate issues and then we can merge this

File: chips/imxrt10xx/src/lpi2c.rs (hunk 3 above)
Comment (brghena) [reply chips/imxrt10xx/src/lpi2c.rs:708]: Moved to issue #3658

File: chips/rp2040/src/gpio.rs (hunk 4 above)
Comment (brghena) [reply chips/rp2040/src/gpio.rs:358]: Moved to issue #3659

**Review by brghena — APPROVED** [review]

**Pull request merged**